lib/Doctrine/Migrations/Tools/Console/Command/ExecuteCommand.php

```diff
@@ -130,7 +132,8 @@ protected function execute(InputInterface $input, OutputInterface $output): int
             : Direction::UP;
 
         $path = $input->getOption('write-sql') ?? getcwd();
-        if (is_string($path) && ! is_writable($path)) {
+
+        if (is_string($path) && (! is_writable($path) && (! is_dir($path) && ! is_writable(dirname($path))))) {
```

Comment:
Maybe this (and the other check in MigrateCommand.php) should be put into a method to make it more readable. Something like `$this->isDirectoryWritable($path)`

Reply:
good idea, done. I would like also to find a way to reduce the code duplication but I really do not know where to put that shared method...

Reply:
I think this should be fine for now.